src/os/ui/config/betatoggleui.js

```diff
@@ -0,0 +1,101 @@
+goog.declareModuleId('os.file.BetaToggleUI');
+
+import Settings from '../../config/settings.js';
+import {ROOT} from '../../os.js';
+import Module from '../module.js';
+
+
+
+/**
+ * The column mapping settings UI directive
+ * @return {angular.Directive}
+ */
+export const directive = () => ({
+  restrict: 'E',
+  replace: true,
+  templateUrl: ROOT + 'views/file/betatoggle.html',
+  controller: Controller,
+  controllerAs: 'ctrl'
+});
+
+/**
+ * The element tag for the directive.
+ * @type {string}
+ */
+export const directiveTag = 'betatoggle';
+
+
+Module.directive(directiveTag, [directive]);
+
+
+/**
+ * Controller for beta features settings.
+ * @unrestricted
+ */
+export class Controller {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @ngInject
+   */
+  constructor($scope) {
+    /**
+     * @type {?angular.Scope}
+     * @protected
+     */
+    this.scope = $scope;
+    this.Settings = Settings.getInstance();
+    // set up an array to store all beta features
+    this['betaFeatureNames'] = [];
+    // iterate through all beta features in the settings.json
+    for (const key in this.Settings.get('betaFeatures')) {
+      // get values for the beta feature
+      const path = this.Settings.get('betaFeatures.' + key);
+      var checkValue = this.Settings.get(path);
+      // store all the beta feature values
+      const betaFeature = new betaFeatureValues(key, path, checkValue);
```

Comment:
I'd consider making a typedef for this instead of a new class type. A typedef will just allow you to make a structured Object that you always know the contents of. And since we don't need to run any functions within the betaFeatureValues type, a typedef is better than a class.